### PR TITLE
Deleting resources on system stop

### LINKED ISSUE #2656: [mesheryctl]  Meshery Operator and its custom resources not deleted on `system stop`
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
`mesheryctl system stop` on Kubernetes does not remove the Meshery Operator and its custom resources.

#### Desired Behavior
Meshery Operator should be removed from the cluster, if present.

#### Implementation
1. Before deleting the Meshery Server pod, invoke the Meshery Server endpoint that manages Meshery Operator in order to delete the Operator.
1. After successful removal of the Operator, proceed with deleting Meshery deployment.

#### Acceptance Tests
1. Operator and any of its CRs should be deleted.
2. Meshery Server and adapters should be deleted.
3. `meshery` namespace should be deleted. 

---

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/layer5io/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Signed-off-by: metonymic-smokey <ahuja.aditi@gmail.com>

**Description**
WIP PR - currently deletes the deployment.

This PR fixes #2656 

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/system/stop.go [hunk 1]
```diff
@@ -141,6 +143,27 @@ func stop() error {
 			return err
 		}
 
+		//getting service endpoint
+		var opts meshkitkube.ServiceOptions
+		opts.Name = "meshery"
+		opts.Namespace = utils.MesheryNamespace
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/stop.go:149]: Perhaps, we should have a central struct that contains this. Thoughts @layer5io/mesheryctl-maintainers?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/stop.go:149]: @kumarabd do the CRs first need to be deleted as a graceful shutdown?

Comment (kumarabd) [reply mesheryctl/internal/cli/root/system/stop.go:149]: Yes they will have to be deleted @lee

Comment (navendu-pottekkat) [reply mesheryctl/internal/cli/root/system/stop.go:149]: @leecalcote we could move "meshery" to a global variable similar to `utils.MesheryNamespace`. But if a `struct` would be cleaner, then we can refactor things to make that change.

File: mesheryctl/internal/cli/root/system/stop.go [hunk 2]
```diff
@@ -141,6 +144,75 @@ func stop() error {
...
+		clientset := client.KubeClient
+		deploymentsClient := clientset.AppsV1().Deployments(utils.MesheryNamespace)
+
+		endpoint, err := meshkitkube.GetServiceEndpoint(context.TODO(), clientset, &opts)
+		if err != nil {
+			return err
+		}
+		_ = endpoint //temporary line
+
+		//delete deployment
```
Comment (kumarabd) [inline mesheryctl/internal/cli/root/system/stop.go:161]: @navendu-pottekkat use manifests which were used for deployment. https://github.com/layer5io/meshery-operator/blob/master/config/manifests/default.yaml

k8s client shudnt be a part of mesheryctl.

**Review by kumarabd — COMMENTED** [review]
ApplyManifest() can be used for deletion.

File: mesheryctl/internal/cli/root/system/stop.go (hunk 2 above)
Comment (kumarabd) [reply mesheryctl/internal/cli/root/system/stop.go:161]: @metonymic-smokey Will you use this with a delete option to delete resources? - https://github.com/layer5io/meshkit/blob/master/utils/kubernetes/apply-manifest.go#L32

You wouldn't need any of the client in the code when u use this

**Comment by leecalcote** [comment]
@metonymic-smokey signal with this is no longer "WIP" and "draft" and you're ready for a review.

**Comment by kumarabd** [comment]
LGTM @metonymic-smokey . @navendu-pottekkat Are we planning on using viper configs for hardcoded values? If yes then the Meshery-operator url and the constants need to be moved. Just an item for the future case.

**Review by kumarabd — DISMISSED** [review]

**Review by navendu-pottekkat — APPROVED** [review]
Thank you @metonymic-smokey . Looks Good.

@leecalcote @abhishekkumkar We can merge this and look into moving all the constants to a central struct as mentioned.

**Comment by leecalcote** [comment]
Awesome. Good work, @metonymic-smokey 😄

**Pull request merged**